src/timeline/TimelineModel.cpp

```diff
@@ -1934,11 +1934,279 @@ TimelineModel::formatPowerLevelEvent(const QString &id)
     if (!event)
         return QString();
 
-    QString user = QString::fromStdString(event->sender);
-    QString name = utils::replaceEmoji(displayName(user));
+    mtx::events::StateEvent<mtx::events::state::PowerLevels> *prevEvent = nullptr;
+    if (!event->unsigned_data.replaces_state.empty()) {
+        auto tempPrevEvent = events.get(event->unsigned_data.replaces_state, event->event_id);
+        if (tempPrevEvent) {
+            prevEvent =
+              std::get_if<mtx::events::StateEvent<mtx::events::state::PowerLevels>>(tempPrevEvent);
+        }
+    }
+
+    QString user        = QString::fromStdString(event->sender);
+    QString sender_name = utils::replaceEmoji(displayName(user));
+    // Get the rooms levels for redactions and powerlevel changes to determine "Administrator" and
+    // "Moderator" powerlevels.
+    auto administrator_power_level = event->content.state_level("m.room.power_levels");
+    auto moderator_power_level     = event->content.redact;
+    auto default_powerlevel        = event->content.users_default;
+    if (!prevEvent)
+        return "";
```

Comment:
You should probably still return something here, i.e. "x setup the power levels"